lib/web/terminal.go

```diff
@@ -405,6 +477,18 @@ func (t *TerminalHandler) windowChange(params *session.TerminalParams) {
 	}
 }
 
+// receivePong is called when server receives a client response to its ping request.
+// It will wait until the channel is able to receive or keep alive request missed
+// count has been maxed.
+func (t *TerminalHandler) receivePong() {
+	select {
+	case t.sendRequestResponse <- struct{}{}:
+		return
+	case <-t.keepAliveContext.Done():
+		return
```

Comment:
I wasn't sure if this was needed. I wasn't sure what scenarios will cause the receiver channel to be blocked, so I put this unblocking case here...